src/include/OpenImageIO/thread.h

```diff
@@ -505,45 +430,288 @@ class mutex_pool
 
 
 
-/// Simple thread group class. This is just as good as boost::thread_group,
-/// for the limited functionality that we use.
+/// Simple thread group class: lets you spawn a group of new threads,
+/// then wait for them to all complete.
 class thread_group {
 public:
     thread_group () {}
-    ~thread_group () {
-        for (size_t i = 0, e = m_threads.size(); i < e; ++i)
-            delete m_threads[i];
-    }
+    ~thread_group () { join_all(); }
+
     void add_thread (thread *t) {
         if (t) {
             lock_guard lock (m_mutex);
-            m_threads.push_back (t);
+            m_threads.emplace_back (t);
         }
     }
-    template<typename FUNC>
-    thread *create_thread (FUNC func) {
-        lock_guard lock (m_mutex);
-        thread *t = new thread (func);
-        m_threads.push_back (t);
+
+    template<typename FUNC, typename... Args>
+    thread *create_thread (FUNC func, Args&&... args) {
+        thread *t = new thread (func, args...);
```

Comment:
Like this?

    template<typename FUNC, typename... Args>
    thread *create_thread (FUNC func, Args&&... args) {
        thread *t = new thread (func, std::forward<Args>(args)...);
        add_thread (t);
        return t;
    }